src/config/groups/keys.cpp

```diff
@@ -1574,6 +1574,15 @@ LIBSESSION_C_API bool groups_keys_key_supplement(
     }
 }
 
+LIBSESSION_EXPORT int groups_keys_current_generation(config_group_keys* conf) {
+    try {
+        return unbox(conf).current_generation();
+    } catch (const std::exception& e) {
+        set_error(conf, e.what());
+        return -1;
+    }
```

Comment:
```suggestion
    return unbox(conf).current_generation();
```

(Continuing the other comment) ... there's nothing in here that can actually throw (it can crash if given an invalid pointer, but that's not something an exception can catch).

Reply:
(pushed)

Reply:
Good to know - I'd just duplicated and modified what the other functions were doing 😂